core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLActionBuilder.kt

```diff
@@ -79,27 +91,127 @@ object GraphQLActionBuilder {
 
 
         }
+
+
         for (elementIntable in table) {
+
             if (elementIntable.tableName == "Mutation" || elementIntable.tableName == "Query") {
-                handleOperation(actionCluster, elementIntable?.tableName, elementIntable?.tableField, elementIntable?.tableType)
+
+                handleOperation(actionCluster, elementIntable.tableField, elementIntable.tableName,
+                        elementIntable.tableType,
+                        elementIntable.kindOfTableType.toString(),
+                        elementIntable.kindOfTableField.toString(),
+                        table,
+                        elementIntable.tableName.toString())
+
             }
         }
 
     }
-
     private fun handleOperation(
             actionCluster: MutableMap<String, Action>,
-            tableName: String?,
-            tableField: String?,
-            tableType: String?
+            methodName: String?,
+            methodType: String?,
+            tableType: String,
+            kindOfTableType: String,
+            kindOfTableField: String,
+            table:MutableList<Table>,
+            tableName: String
     ) {
+        if(methodName == null){
+            //TODO log warn
+            return;
+        }
+        if(methodType == null){
+            //TODO log warn
+            return;
+        }
+        val type = when{
+            methodType.equals("QUERY",true) -> GQMethodType.QUERY
+            methodType.equals("MUTATION",true) -> GQMethodType.MUTATION
+            else -> {
+                //TODO log warn
+                return
+            }
+        }
 
+        val actionId = "$methodName${idGenerator.incrementAndGet()}"
 
-        val actionId = "$tableName$tableField${idGenerator.incrementAndGet()}"
-        val action = GraphQLAction(actionId, tableName, tableField, tableType)
+        //TODO populate
+
+        val params = extractParams(methodName, methodType, tableType, kindOfTableType, kindOfTableField, table, tableName)
```

Comment:
once we compute `type`, why passing done `methodType`?

Reply:
Fixed